### DISCUSSION TITLE
Using targets in a bioinformatics workflow

### DISCUSSION THREAD

**Discussion by joelnitta** [ropensci/targets#710] [General]
Hi everyone! Apologies for the self-promotion, but I wrote a blogpost on combining targets with Conda, Docker, and renv to run bioinformatics workflows (similar to [nextflow](https://www.nextflow.io/), [CWL](https://www.commonwl.org/), etc). I hope it is useful for some in this community!

https://joelnitta.com/r-bioinfo-flow

**Comment by wlandau** [comment]
That’s awesome, Joel! Thanks for sharing! Great tip about babelwhale within a target, I will start recommending that pattern. Compared to Docker futures, your way looks easier to combine with other kinds of parallelism among targets, e.g. tar_make_future() with future.callr. I like how your post decouples those elements.

**Reply by joelnitta** [reply]
Thanks for the feedback! Honestly, docker futures didn't even occur to me... I vaguely remember seeing that somewhere, but now I can't find it. Do you have a link? 

Anyways yes, I'm quite pleased with how well babelwhale works in this context. I'm even thinking of creating a package to wrap functions around babelwhale calls for commonly used bioinformatics tools. As you pointed out, this approach works nicely with `tar_make_future()` so no changes need to be made to the functions themselves to run them in parallel.

**Reply by wlandau** [reply]
I don’t remember where it is in the {future} documentation, but there is a small example at https://github.com/HenrikBengtsson/future/issues/174.

**Reply by joelnitta** [reply]
I see, thanks! Also, I remember where I'd previously seen this: one of our earlier discussions: https://github.com/ropensci/targets/discussions/446. It even mentions my now (finally!) completed blog post.

Feels kind of like googling an error and finding your own answer...
